contrib/labstack/echo/echotrace.go

```diff
@@ -54,12 +56,35 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			// serve the request to the next middleware
 			err := next(c)
 			if err != nil {
-				span.SetTag(ext.Error, err)
-				// invokes the registered HTTP error handler
-				c.Error(err)
+				// It is impossible to determine what the final status code of a request is in echo.
+				// This is the best we can do.
+				switch err := err.(type) {
+				case *echo.HTTPError:
+					if cfg.isStatusError(err.Code) {
+						// mark 5xx server error
+						span.SetTag(ext.Error, err)
+					}
+					span.SetTag(ext.HTTPCode, strconv.Itoa(err.Code))
+				default:
+					// Any non-HTTPError errors appear as 5xx errors.
+					if cfg.isStatusError(500) {
+						span.SetTag(ext.Error, err)
+					}
+					span.SetTag(ext.HTTPCode, "500")
+				}
+			} else if status := c.Response().Status; status > 0 {
+				if cfg.isStatusError(status) {
+					// mark 5xx server error
```

Comment:
These comments are confusing...

Another Q: can't we share this code somehow between the two integrations? I haven't looked closely. In case there are subtleties which block that, ignore this question...

Reply:
I looked at this briefly and it might be possible, but it was a little gross because of the different echo types.
Let's put a pin in that for the future.